code/modules/antagonists/heretic/items/hunter_rifle.dm

```diff
@@ -0,0 +1,143 @@
+// The Lionhunter, a gun for heretics
+// The ammo it uses takes time to "charge" before firing,
+// releasing a homing, very damaging projectile
+/obj/item/gun/ballistic/rifle/lionhunter
+	name = "\improper Lionhunter's Rifle"
+	desc = "An antique looking rifle that looks immaculate despite being clearly very old."
+	slot_flags = ITEM_SLOT_BACK
+	icon_state = "moistprime"
+	inhand_icon_state = "moistprime"
+	worn_icon_state = "moistprime"
+	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/lionhunter
+	fire_sound = 'sound/weapons/gun/sniper/shot.ogg'
+	zoomable = TRUE
+	zoom_amt = 5
+	zoom_out_amt = 3
+
+/obj/item/ammo_box/magazine/internal/boltaction/lionhunter
+	name = "lionhunter rifle internal magazine"
+	ammo_type = /obj/item/ammo_casing/a762/lionhunter
+	caliber = CALIBER_A762
+	max_ammo = 3
+	multiload = TRUE
+
+/obj/item/ammo_casing/a762/lionhunter
+	projectile_type = /obj/projectile/bullet/a762/lionhunter
+	/// Whether we're currently aiming this casing at something
+	var/currently_aiming = FALSE
+	/// How many seconds it takes to aim per tile of distance between the target
+	var/seconds_per_distance = 0.5
+	/// The minimum distance required to gain a damage bonus from aiming
+	var/min_distance = 4
+
+/obj/item/ammo_casing/a762/lionhunter/fire_casing(atom/target, mob/living/user, params, distro, quiet, zone_override, spread, atom/fired_from)
+	if(!loaded_projectile)
+		return
+	if(!check_fire(target, user))
+		return
+
+	return ..()
+
+/// Checks if we can successfully fire our projectile.
+/obj/item/ammo_casing/a762/lionhunter/proc/check_fire(atom/target, mob/living/user)
+	// In case someone puts this in turrets or something wacky, just fire like normal
+	if(!iscarbon(user) || !istype(loc, /obj/item/gun/ballistic/rifle/lionhunter))
+		return TRUE
+
+	if(currently_aiming)
+		user.balloon_alert(user, "already aiming!")
+		return FALSE
+
+	var/distance = get_dist(user, target)
+	var/fire_time = min((distance SECONDS) * seconds_per_distance, 10 SECONDS)
```

Comment:
can you remove the 'distance SECONDS' and just change it to be directly distance*10?
I don't think the time defines make it any more readable than just doing a direct multiplication